### PR TITLE
Update Dependencies - create-react-app

### PR DESCRIPTION
**Pull request by JackLivio**
Recreated project using node version 12 and the [create-react-app command](https://github.com/facebook/create-react-app)

New build steps

```
nvm use 12
rm -rf node_modules
rm package-lock.json
npm install

npm run build
cd build
chromium index.html

```

### CODE REVIEW

File: src/js/reducers.js [hunk 1]
```diff
@@ -298,18 +296,14 @@ function ui(state = {}, action) {
...
-            var app = newState[action.appID] ? newState[action.appID] : newAppState()
             newState[action.appID] = app
             app.menu = deleteCommand(app.menu, action.cmdID)
             return newState
         case Actions.ADD_SUB_MENU:
-            var newState = { ...state }
-            var app = newState[action.appID] ? newState[action.appID] : newAppState()
             newState[action.appID] = app
-            var menu = app.menu
+            menu = app.menu
```
Comment (iCollin) [inline src/js/reducers.js]: this line can be removed, it is assigned to the same above

File: src/js/reducers.js [hunk 2]
```diff
@@ -323,62 +317,48 @@ function ui(state = {}, action) {
                 menu.push(menuItem);
             return newState
         case Actions.DELETE_SUB_MENU:
-            var newState = { ...state }
-            var app = newState[action.appID] ? newState[action.appID] : newAppState()
             newState[action.appID] = app
-            var menu = app.menu
-            var i = menu.findIndex((command) => {
+            menu = app.menu
```
Comment (iCollin) [inline src/js/reducers.js]: this line can be removed, it is assigned to the same above

File: src/js/reducers.js [hunk 3]
```diff
@@ -241,13 +240,16 @@ function deleteCommand(commands, cmdID) {
...
+    var app = newState[action.appID] ? newState[action.appID] : newAppState();
+    var menu = app.menu;
+    var menuItem = null;
+    var i = 0;
     switch (action.type) {
-        case Actions.SHOW:
-            var newState = { ...state }
-            var app = newState[action.appID] ? newState[action.appID] : newAppState()
+        case Actions.SHOW:           
             newState[action.appID] = app
```
Comment (iCollin) [inline src/js/reducers.js]: could `newState[action.appID] = app` be moved above the switch? it's in many cases. depends on mutability i guess,  if i change app later, does it change `newState[action.appID] ` (i think yes)

File: src/js/reducers.js [hunk 4]
```diff
@@ -456,27 +435,25 @@ function ui(state = {}, action) {
...
             return newState
-        case Actions.REGISTER_APPLICATION:
-            var newState = { ...state }         
+        case Actions.REGISTER_APPLICATION:        
             if (!newState[action.appID]) {
               newState[action.appID] = newAppState()
             }
-            var app = newState[action.appID]
-            if (app.displayLayout == null) {
+            app = newState[action.appID]
```
Comment (iCollin) [inline src/js/reducers.js]: i'm pretty sure this line and the three above it are unnecessary with new changes to code before the switch

File: src/js/reducers.js [hunk 5]
```diff
@@ -456,27 +435,25 @@ function ui(state = {}, action) {
...
+            if (app.displayLayout === null) {
```
Comment (iCollin) [inline src/js/reducers.js]: I still think nullchecks should use `==` or `!obj` to also catch `undefined`

File: src/logo.svg [hunk 6]
```diff
@@ -0,0 +1 @@
....23-.12s-3.58,0-5.33.12C88.45,50.57,90.2,48.48,92,46.56Zm-22.18-12A4.82,4.82,0,0,1,72.29,34a13.11,13.11,0,0,1,5.19,1.31,39.07,39.07,0,0,1,10,6.78l1.17,1.07a98.49,98.49,0,0,0-8.16,10,98.88,98.88,0,0,0-12.65,2.06c-.44-1.94-.8-3.84-1-5.67C65.8,42,67,36.24,69.81,34.6ZM64.53,85.26a58.75,58.75,0,0,1-5.54-2c-7.1-3-11.5-6.85-11.5-10.14S51.91,66,59,63.05a57.63,57.63,0,0,1,5.44-1.94A97.93,97.93,0,0,0,69,73.25,98.72,98.72,0,0,0,64.53,85.26Zm23,19.1c-8.19,7.33-13.31,8.11-15.14,8.11a4.69,4.69,0,0,1-2.42-.58c-2.86-1.63-4-7.38-3.09-15a56.07,56.07,0,0,1,1-5.78,99.51,99.51,0,0,0,12.58,2.06,97.17,97.17,0,0,0,8.24,10.08Zm4.57-4.51c-1.8-1.92-3.57-4-5.31-6.23,1.72.08,3.47.12,5.24.12s3.58,0,5.33-.11C95.63,95.85,93.87,97.93,92.09,99.85Zm22.18,12a4.82,4.82,0,0,1-2.48.59c-1.82,0-7-.8-15.16-8.1l-1.17-1.07a98.44,98.44,0,0,0,8.15-10,97,97,0,0,0,12.66-2.06c.44,1.94.79,3.84,1,5.67C118.27,104.42,117.12,110.18,114.27,111.81Zm10.8-28.44c-1.71.7-3.52,1.35-5.44,1.93a98.54,98.54,0,0,0-4.57-12.14,98.1,98.1,0,0,0,4.49-12,58.75,58.75,0,0,1,5.54,2c7.09,3,11.5,6.85,11.49,10.14S132.17,80.42,125.07,83.37ZM92,82.39a9.18,9.18,0,1,0-9.17-9.19A9.17,9.17,0,0,0,92,82.39Zm-69.32,54.5V26.2L17.89,31V141.66H139.94l4.78-4.77Z"/></svg>
```
Comment (iCollin) [inline src/logo.svg]: where do we use this? is it necessary?

File: src/js/containers/Metadata.js [hunk 7]
```diff
@@ -40,7 +40,10 @@ const mapStateToProps = (state) => {
                 case "mainField4":
                     props.mainField4 = metadata.showStrings[fieldName]
                     break
+                default:
+                    break;
             }
+            return true;
```
Comment (iCollin) [inline src/js/containers/Metadata.js]: this seems like it should be a foreach and not a map

File: src/App.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```
Comment (iCollin) [inline src/App.js]: can we remove this file?

File: src/App.css [hunk 9]
```diff
@@ -0,0 +1,22 @@
+.App {
```
Comment (iCollin) [inline src/App.css]: can we remove this file?

File: src/App.test.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (iCollin) [inline src/App.test.js]: can we remove this file?

File: README.md [hunk 11]
```diff
@@ -1,373 +1,68 @@
-# Getting Started
```
Comment (iCollin) [inline README.md]: um, should we keep some of the old README??

File: src/js/Controllers/NavController.js [hunk 12]
```diff
@@ -9,20 +9,21 @@ class NavController {
     }
     handleRPC(rpc) {
         let methodName = rpc.method.split(".")[1]
+        var message = "";
```
[the remark this answers is not in the document]
Comment (JackLivio) [reply src/js/Controllers/NavController.js]: The linter did not like redefining message even though it was in a block scope. I think this looks cleaner than cramming the message as a parameter.

File: src/js/reducers.js (hunk 3 above)
Comment (JackLivio) [reply src/js/reducers.js]: Good idea, I think you are correct about the reference

File: src/js/Controllers/NavController.js (hunk 12 above)
[the remark this answers is not in the document]
Comment (iCollin) [reply src/js/Controllers/NavController.js]: Sounds good to me. Weird linter behavior imo

File: README.md (hunk 11 above)
Comment (JackLivio) [reply README.md]: Yes

File: src/js/ControlBarItem.js [hunk 13]
```diff
@@ -16,6 +20,23 @@ export default class ControlBar extends React.Component {
...
+            case "iconSeekRight":
+                return <IconSeekRight/>
+            case "iconPlay":
+                return <IconPlay/>
+            case "iconPause":
+                return <IconPause/>
+            case "iconPlayPause":
+                return <IconPlayPause/>
+            case "iconStop":
+                return <IconStop/>
```
Comment (iCollin) [inline src/js/ControlBarItem.js]: Should add a default case to return null here (appease linter)

File: src/js/Templates/Shared/SoftButtons.js [hunk 14]
```diff
@@ -49,7 +45,7 @@ class SoftButtonsBody extends React.Component {
         var cssColorStyle = Object.assign(primaryStyle ? primaryStyle : {},
```
Comment (iCollin) [inline src/js/Templates/Shared/SoftButtons.js]: Should remove `var id=0` from above, it is never used (appease linter)

**Comment by iCollin** [comment]
the README should mention that you now have to build the HMI before you can run it
also where it says chromium index.html it should be build/index.html

do we also want to fix some of the compile warnings now?

**Pull request merged**